### ISSUE TITLE
Adaptive restarts & continuous actions

### ISSUE THREAD

**Issue by bigaidream** [bigaidream-projects/qan#30]
It has been shown in [SGDR: Stochastic Gradient Descent with Restarts] that restarts can accelerate convergence dramatically. It seems much better than ADAM or other similar learning rate schedulers. 

In this situation, we only have 2 actions: keep `decreasing` or `restart` to a certain level (manually specified or tuned by Bayesian optimization). However, the number of `restart` will be much smaller than `decrease`. This is like a semi-Markov decision problem (suggested by Miao Liu), where the different actions may take different amount of time to execute. http://www-anw.cs.umass.edu/~barto/courses/cs687/Sutton-Precup-Singh-AIJ99.pdf

We might want to consider the restart action just takes multiple time steps, and decreasing learning rate as another action (or set of actions) with multiple time steps to execute (or set of different time scales). That way, the action sample size might be reduced (that is the original single-step sized actions becomes temporarily extended actions). 

A simple strategy would be to define `decrease` as an action over multiple iterations (i.e. decrease by 10% for 100 iterations); `restart` as an action to return to a certain learning rate and decrease it by 10% for 100 iterations.

**Comment by bigaidream** [comment]
In fact, we can simply use `continuous` actions rather than `discrete` ones.

**Comment by bigaidream** [comment]
Not practical at this moment.
